pkg/docker/sandbox_util.go

```diff
@@ -0,0 +1,201 @@
+package docker
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/docker/docker/client"
+
+	"github.com/enescakir/emoji"
+
+	cmdUtil "github.com/flyteorg/flytectl/pkg/commandutils"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/mount"
+	"github.com/docker/go-connections/nat"
+	f "github.com/flyteorg/flytectl/pkg/filesystemutils"
+)
+
+var (
+	Kubeconfig              = f.FilePathJoin(f.UserHomeDir(), ".flyte", "k3s", "k3s.yaml")
+	FlytectlConfig          = f.FilePathJoin(f.UserHomeDir(), ".flyte", "config-sandbox.yaml")
+	SuccessMessage          = "Flyte is ready! Flyte UI is available at http://localhost:30081/console"
+	ImageName               = "cr.flyte.org/flyteorg/flyte-sandbox:dind"
+	FlyteSandboxClusterName = "flyte-sandbox"
+	Environment             = []string{"SANDBOX=1", "KUBERNETES_API_PORT=30086", "FLYTE_HOST=localhost:30081", "FLYTE_AWS_ENDPOINT=http://localhost:30084"}
+	FlyteSnackDir           = "/usr/src"
+	K3sDir                  = "/etc/rancher/"
+	Client                  Docker
+	Volumes                 = []mount.Mount{
+		{
+			Type:   mount.TypeBind,
+			Source: f.FilePathJoin(f.UserHomeDir(), ".flyte"),
+			Target: K3sDir,
+		},
+	}
+)
+
+// SetupFlyteDir will create .flyte dir if not exist
+func SetupFlyteDir() error {
+	if err := os.MkdirAll(f.FilePathJoin(f.UserHomeDir(), ".flyte"), 0755); err != nil {
+		return err
+	}
+	return nil
+}
+
+// GetFlyteSandboxConfig download the flyte sandbox config
+func GetFlyteSandboxConfig() error {
+	response, err := http.Get("https://raw.githubusercontent.com/flyteorg/flytectl/master/config.yaml")
+	if err != nil {
+		return err
+	}
+	defer response.Body.Close()
+
+	data, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return err
+	}
+
+	_ = ioutil.WriteFile(FlytectlConfig, data, 0600)
```

Comment:
If the client has already run the sandbox and updated his config.yaml this would overwrite it.

Reply:
yes, but I am not using their default config. It will create a new file `~/.flyte/config-sandbox.yaml` and then set the env variable.

Reply:
Is that the same k3.yaml too ?